### PR TITLE
RM-7841: Replace Rhino.Mocks with Moq in Security Manager

### PR DESCRIPTION
**Pull request by derlistigelurch**

### CODE REVIEW

**Review by iam-mholle — CHANGES_REQUESTED** [review]

File: SecurityManager/Clients.Web.Test.SetupDB/DBUtility.cs [hunk 1]
```diff
@@ -27,7 +27,6 @@
 
 namespace Remotion.SecurityManager.Clients.Web.Test.SetupDB
 {
-
```
[the remark this answers is not in the document]
Comment (derlistigelurch) [reply SecurityManager/Clients.Web.Test.SetupDB/DBUtility.cs:30]: uuups! that was a manual change, I thought the rewriter accidentally inserted a new line and therefore I deleted it

File: SecurityManager/Core.UnitTests/AclTools/Expansion/AclExpansionHtmlWriterTest.cs [hunk 2]
```diff
@@ -676,27 +676,26 @@ public void WriteTableBody_ProcessClassClassNodeKeyIsNullTest ()
...
-        AclExpansionTreeNode<AclExpansionEntry, AclExpansionEntry>>> (null, nrLeafNodes, aclExpansionTreeChildrenMock);
-      aclExpansionTreeNodeMock.Expect(x => x.NumberLeafNodes).Return(nrLeafNodes);
-      aclExpansionTreeNodeMock.Expect(x => x.Children).Return(aclExpansionTreeChildrenMock);
+      var aclExpansionHtmlWriterImplementation = new Mock<AclExpansionHtmlWriterImplementation> (TextWriter.Null, true, new AclExpansionHtmlWriterSettings());
+      aclExpansionHtmlWriterImplementation.Setup (x => x.WriteTableDataWithRowCount (AclToolsExpansion.NO_CLASSES_DEFINED_Text, nrLeafNodes)).Verifiable();
 
-      var aclExpansionHtmlWriterImplementation = mocks.DynamicMock<AclExpansionHtmlWriterImplementation>(TextWriter.Null,true,new AclExpansionHtmlWriterSettings());
-      aclExpansionHtmlWriterImplementation.Expect(x => x.WriteTableDataWithRowCount (AclToolsExpansion.NO_CLASSES_DEFINED_Text, nrLeafNodes));
+      var aclExpansionHtmlWriterMock = new Mock<AclExpansionHtmlWriter> (
+          aclExpansionHtmlWriterImplementation.Object, new AclExpansionHtmlWriterSettings()) { CallBase = true };
```
Comment (iam-mholle) [inline SecurityManager/Core.UnitTests/AclTools/Expansion/AclExpansionHtmlWriterTest.cs:690]:
```suggestion
          aclExpansionHtmlWriterImplementation.Object,
          new AclExpansionHtmlWriterSettings())
          { CallBase = true };
```

File: SecurityManager/Core.UnitTests/Domain/AccessControl/AccessEvaluation/SecurityTokenBuilderTest.cs [hunk 3]
```diff
@@ -592,6 +592,7 @@ public void Create_WithValidOwningTenant ()
       SecurityContext context = CreateContext();
       ISecurityPrincipal user = CreateTestPrincipal();
 
+
```
[the remark this answers is not in the document]
Comment (derlistigelurch) [reply SecurityManager/Core.UnitTests/Domain/AccessControl/AccessEvaluation/SecurityTokenBuilderTest.cs]: there was a newline which was accidentaly deleted, and i inserted in the wrong line🤫

**Review by iam-mholle — APPROVED** [review]

**Review by MichaelKetting — CHANGES_REQUESTED** [review]

File: SecurityManager/Core.UnitTests/Domain/OrganizationalStructure/GroupTests/BindableObjectImplementation.cs [hunk 4]
```diff
@@ -147,50 +147,50 @@ public void GetPropertyDefinitions_CheckForUniqueIdentifier ()
...
       Assert.That (groupTypeProperty, Is.Not.Null);
 
       var group = CreateGroup();
-      var expected = new[] { MockRepository.GenerateStub<IBusinessObject> () };
+      var expected = new[] { new Mock<IBusinessObject>().Object };
 
-      searchServiceStub.Stub (stub => stub.SupportsProperty (groupTypeProperty)).Return (true);
-      searchServiceStub.Stub (stub => stub.Search (group, groupTypeProperty, args)).Return (expected);
+      searchServiceStub.Setup (stub => stub.SupportsProperty (groupTypeProperty)).Returns (true);
+      searchServiceStub.Setup (stub => stub.Search (group, groupTypeProperty, args.Object)).Returns ((IBusinessObject[]) expected);
```
[the remark this answers is not in the document]
Comment (derlistigelurch) [reply SecurityManager/Core.UnitTests/Domain/OrganizationalStructure/GroupTests/BindableObjectImplementation.cs]: no the cast is not needed anymore, therefore it was deleted in the latest commit

File: SecurityManager/Core.UnitTests/Domain/SearchInfrastructure/OrganizationalStructure/RolePropertiesSearchServiceTests/SearchPositionWithSecurityChecks.cs [hunk 5]
```diff
@@ -99,17 +97,18 @@ public override void TearDown ()
...
+      SetupResultSecurityProviderGetAccessForPosition (Delegation.Disabled, principalStub.Object);
       Group rootGroup = _expectedRootGroupHandle.GetObject();
-      _mocks.ReplayAll();
 
       var positions = _searchService.Search (null, _positionProperty, CreateSearchArguments (rootGroup));
 
-      _mocks.VerifyAll();
+      _stubFunctionalSecurityStrategy.Verify();
+      _mockPrincipalProvider.Verify();
+      principalStub.Verify();
```
[the remark this answers is not in the document]
Comment (derlistigelurch) [reply SecurityManager/Core.UnitTests/Domain/SearchInfrastructure/OrganizationalStructure/RolePropertiesSearchServiceTests/SearchPositionWithSecurityChecks.cs]: this is a bug in the rewriter and will be fixed soon, for now i deleted the stub verification manually

**Review by MichaelKetting — APPROVED** [review]

**Pull request merged**